Foundation/NSConcreteValue.swift

```diff
@@ -11,7 +11,7 @@ import CoreFoundation
 
 #if os(OSX) || os(iOS)
     import Darwin
-#elseif os(Linux)
+#elseif os(Linux) || IS_CYGWIN
```

Comment:
Some headers ```aio.h, cpio.h, fmtmsg.h, nl_types.h, ulimit.h, ``` are not exist in Cygwin. If we can not distinguish Cygwin from Linux, the Swift code on Cygwin can not know if the Glibc function is implemented.

Current swift compiler is implemented as ```os(Windows)``` is true for ```Windows (MSVC)``` and ```Cygwin```. Since ```os()``` is exclusive, ```os(Linux)``` is false in Cygwin. There was a not finished (holded) discussion. ([swift-evolution ](https://lists.swift.org/pipermail/swift-evolution/Week-of-Mon-20160502/thread.html#16381)).
I think ```IS_CYGWIN``` is best until the discussion.